src/net.cpp

```diff
@@ -3131,6 +3263,58 @@ CNode::~CNode()
     GetNodeSignals().FinalizeNode(GetId());
 }
 
+void CNode::PrioritizeSendMsg(CNode *pnode)
+{
+    LOCK(cs_PrioritySendQ);
+    vPrioritySendQ.push_back(CNodeRef(pnode));
+
+    // if there are more priority messages than the one we just added then find out how many
+    // are currently from this peer so we can order them by time of entry.
+    uint32_t nItemsPriority = 0;
+    if (vPrioritySendQ.size() >= 2)
+    {
+        for (auto &noderef : vPrioritySendQ)
+        {
+            if (noderef.get() == pnode)
+                nItemsPriority++;
+        }
+    }
+    else
+        nItemsPriority = 1;
+
+    // Move the range of priority msg's forward and place the new priority message (pri3) at the
```

Comment:
I know what you're saying....this is the first approach I took...It seems so simple, have two separate queues, just like the priority recv messages...but it's much more complicated when it comes time to actually send the data in socketsenddata. It's been 6 months since I've looked at it but from what I remember, it would have needed some bigger and more difficult changes to socket send data...and I can't remember why so I'll have to take a closer look again.   Maybe we can get together next week sometime and discuss on Keybase in the meantime I'll refresh my memory.

Reply:
I think I might have another approach...you're right, the current set of code works, it is ugly and hard to follow...I'm going to just create a different set of counters, really is just sendoffset we need to track, but I"ll need a new socketsenddata() , will call it prioritysocketsenddata()...that's where I was getting stuck before , trying to wed the two together. Will have take a couple of days to work it out and see where it goes.